src/textual/app.py

```diff
@@ -1668,6 +1671,11 @@ def _register(
                 if widget.children:
                     self._register(widget, *widget.children)
                 apply_stylesheet(widget)
+
+        if not self._running:
```

Comment:
I get that if the app is not running we don't want to return widgets to be awaited... but if I am in the method that registers widgets, isn't this a weird place to check if the app is running, right before returning..?

Reply:
It's mainly for the benefit of tests which don't run an app. We still want to check for duplicate IDs and confirm the DOM state has been set up.